Clarified documentation about checking Publishers, Subscribers and AdvertiseServices.

Bardo91 opened the pull request:
As suggested in [https://github.com/ros/ros_comm/issues/582], clarified methodology of checking if Publisher, Subscribers and AdvertiseServices are valid. Added definition of objects in order not to confuse old "handle" variable of type Publisher/Subscriber/AdvertiseSevice with NodeHandle type.

Commit:
Clarified documentation about checking of Publishers, Subscribers and AdvertiseServices.
As suggested in [https://github.com/ros/ros_comm/issues/582#issuecomment-83037782], clarified methodology of checking if Publisher, Subscribers and AdvertiseServices are valid. Added definition of objects in order not to confuse old "handle" variable of type Publisher/Subscriber/AdvertiseSevice with NodeHandle type.

ros-pull-request-builder commented:
Can one of the admins verify this patch?

dirk-thomas commented on clients/roscpp/include/ros/node_handle.h:
```diff
@@ -289,7 +289,9 @@ namespace ros
-if (handle)
+ros::NodeHandle nodeHandle;
+ros::publisher pub = nodeHandle.advertise<std_msgs::Empty>("my_topic", 1, (ros::SubscriberStatusCallback)callback);
+if (pub)	// Enter if publisher is valid
```
Please replace the tab before the `//` with two spaces. Same for all the other blocks.

Bardo91 replied:
Done in next commit.

Commit:
Replaced tabs with two spaces.
In node_handle.h documentation, replaced tabs with two spaces in every modified blocks.

dirk-thomas commented on clients/roscpp/include/ros/node_handle.h:
```diff
@@ -810,7 +848,14 @@ ros::ServiceServer service = handle.advertiseService("my_service", &Foo::callbac
+ros::NodeHandle nodeHandle;
+Foo foo_object;
+ros::ServiceServer service = nodeHandle.advertiseService("my_service", &Foo::callback, &foo_object);
+if (service)  // Enter if advertise service is valid
```
Spelling: `Enter if advertised service is valid` (adding a `d` after `advertise`).

Commit:
Fixed spelling.
In node_handle.h fixed spelling ("advertise" to "advertised") in doc blocks

dirk-thomas commented:
Thank you, squashed and merged in cc9d91a3533d027621160ea2e23b55b6ed1f44f8.
